src/components/blocks/field.js

```diff
@@ -235,6 +235,51 @@ class Field extends ProcessSchema {
         return false;
     }
 
+    addRefToNonActiveValue(ref) {
+        if(ref.from_node && !this.nonActiveValue.refs.some(item => item.from_node === ref.from_node)){
+            this.nonActiveValue.refs.push(ref);
+        }
+        else if(ref.from_argument && !this.nonActiveValue.callbackRefs.some(item => item.from_argument === ref.from_argument)){
+            this.nonActiveValue.callbackRefs.push(ref);
+        }
+    }
+
+    addRefsToNonActiveValue(value){
+        for(let key in value){
+            if(value[key] && typeof value[key] === "object"){
+                this.addRefsToNonActiveValue(value[key]);
+            }
+        }
+        if(Field.isRef(value)){
+            this.addRefToNonActiveValue(value);
+        }
+    }
+
+    findInValue(target, value) {
+        var found = false;
+        for(let key in value){
+            if(value[key] && typeof value[key] === "object"){
+                found = this.findInValue(target, value[key]);
+                if(found)
+                    return found;
+            }
+            if(target[key] && value[key] === target[key]){
+                return true;
+            }
+        }
+        //false
+        return found;
+    }
+
+    dashNonActiveEdges() {
+        let edges = this.getEdges();
+        edges.forEach((edge) => {
```

Comment:
Should be `for(var edge of edges) {`. Otherwise it's not clear whether it's async or not and thus it could happen that redrawing would be done before the loop ends.